Lib/poplib.py

```diff
@@ -217,6 +218,51 @@ def pass_(self, pswd):
         """
         return self._shortcmd('PASS %s' % pswd)
 
+    def auth(self, mechanism, authobject=None, initial_response=None):
+        """Authenticate to the POP3 server using the AUTH command (RFC 5034).
+
+        Result is 'response'.
+        """
+        if authobject is not None and initial_response is not None:
+            raise ValueError('authobject and initial_response are mutually exclusive')
+
+        if initial_response is not None:
+            if isinstance(initial_response, str):
+                initial_response = initial_response.encode(self.encoding)
+            b64 = base64.b64encode(initial_response).decode('ascii')
```

Comment:
Technically per the RFC we should check how long this is going to make the AUTH command, and not do the initial response it if would exceed 255 bytes.